sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ConfigProperties.java

```diff
@@ -129,4 +181,40 @@ private static ConfigurationException newInvalidPropertyException(
         .filter(s -> !s.isEmpty())
         .collect(Collectors.toList());
   }
+
+  /** Returns the TimeUnit associated with a unit string. Defaults to milliseconds. */
+  private static TimeUnit getDurationUnit(String unitString) {
+    // If statement in likely-to-be-used order.
+    if ("s".equals(unitString)) {
+      return TimeUnit.SECONDS;
+    } else if ("ms".equals(unitString)) {
+      return TimeUnit.MILLISECONDS;
+    } else if ("m".equals(unitString)) {
+      return TimeUnit.MINUTES;
+    } else if ("h".equals(unitString)) {
+      return TimeUnit.HOURS;
+    } else if ("d".equals(unitString)) {
+      return TimeUnit.DAYS;
+    } else if ("".equals(unitString)) {
+      return TimeUnit.MILLISECONDS;
+    }
+    throw new ConfigurationException("Invalid duration string, found: " + unitString);
+  }
+
+  /**
+   * Fragments the 'units' portion of a config value from the 'value' portion.
+   *
+   * <p>E.g. "1ms" would return the string "ms".
+   */
+  private static String getUnitString(String rawValue) {
+    int i = rawValue.length() - 1;
+    while (i >= 0) {
+      char c = rawValue.charAt(i);
+      if (Character.isDigit(c)) {
+        break;
+      }
+      i -= 1;
+    }
+    return rawValue.substring(i + 1);
```

Comment:
Two points:

1. Reverse parsing is common tactic.   If you'd like I can fix up readability.  Your code finds the first non-digit, my code find the last digit.  Renaming "i" to "lastDigitIndex" should clarify that
2. Reverse-parsing is a LOT more efficient here.  Usually my while loop is a no-op, and it should iterate over far less characters.  I'd hate to swap this in case someone does try to use this where performance matters (or copy-pasta is somewhere else without understanding that nuance).